packages/oui-file/src/file.controller.js

```diff
@@ -0,0 +1,204 @@
+        this.$window = $window;
+
+        this.translations = ouiFileConfiguration.translations;
+        this.units = this.$filter("orderBy")(ouiFileConfiguration.units, "-size");
+    }
+
+    checkFileValidity (file) {
+        if (file) {
+            file.errors = {};
+
+            // Check maxsize
+            if (this.maxsize && file.size > this.maxsize) {
+                file.errors.maxsize = true;
+            }
+
+            // Set form validation
+            if (this.form && this.form[this.name]) {
+                this.form[this.name].$setValidity("maxsize", !file.errors.maxsize);
+                this.form[this.name].$setDirty();
+            }
+
+            // Clean errors
+            if (isEmpty(file.errors)) {
+                delete file.errors;
+            }
+        }
+
+        return file;
+    }
+
+    loadFilePreview (file) {
+        // Load preview only if image
+        if (this.preview && !file.errors && file.type && file.type.search(/^image\//) !== -1) {
+            file.loading = true;
+
+            file.reader = new this.$window.FileReader();
+            file.reader.readAsDataURL(file);
+            file.reader.onload = () => {
+                file.preview = `url("${file.reader.result}")`;
+
+                // Clean file object
+                delete file.loading;
+                delete file.reader;
+
+                this.$scope.$apply();
+            };
+        }
+
+        return file;
+    }
+
+    addFile (file) {
+        this.model = [this.checkFileValidity(file)];
+        this.onSelect({ modelValue: this.model });
+        this.$scope.$apply();
+
+        // Set back focus on fake selector
+        this.$element[0].querySelector(".oui-file-selector__label").focus();
+    }
+
+    addFiles (files) {
+        if (!this.model) {
+            this.model = [];
+        }
+
+        if (angular.isArray(files)) {
+            files.forEach((file) => {
+                // Check for duplicate before adding
+                if (!find(this.model, (item) => file.name === item.name)) {
+                    this.checkFileValidity(file);
+                    this.loadFilePreview(file);
+                    this.model.push(file);
+                }
+            });
+
+            this.onSelect({ modelValue: this.model });
+            this.$scope.$apply();
+        }
+    }
+
+    removeFile (file) {
+        if (angular.isArray(this.model)) {
+            remove(this.model, (item) => item === file);
+        }
+    }
+
+    resetFile () {
+        this.model = undefined;
+        this.fileSelector[0].value = "";
+
+        if (this.form && this.form[this.name]) {
+            this.form[this.name].$setValidity("maxsize", true);
+        }
+    }
+
+    openFileSelector () {
+        // triggerHandler("click") don't work here
+        this.fileSelector[0].click();
+    }
+
+    static getFileName (file) {
+        // Remove file extension from name
+        const parts = file.name.split(".");
```

Comment:
```suggestion
        return file.name.replace(/\.[^/.]+$/, "");
```

Reply:
But what if the file doesn't have an extension ?

Reply:
I want to upload my `.gitignore` file

Reply:
In your case, `gitignore` is the extension. And the filename is ` `.
But I thought about one thing. It will not work with a `.tag.gz` file. It'll only take `.gz` has an extension. I'll check to modify that

Reply:
Did you find a solution ?

Reply:
It seems there is no real solution, even from the MIME. So it'll stay as is, since it's real purpose is for ellipsis. But I'll need to cover files like `test` or `.test`.